src/stellar_types.h

```diff
@@ -221,11 +224,57 @@ typedef struct {
 } MuxedAccount;
 
 typedef struct {
-    AssetType type;
     const char *assetCode;
     AccountID issuer;
+} AlphaNum4;
+
+typedef struct {
+    const char *assetCode;
+    AccountID issuer;
+} AlphaNum12;
+
+typedef struct {
+    AssetType type;
+    union {
+        AlphaNum4 alphaNum4;
+        AlphaNum12 alphaNum12;
+    };
 } Asset;
 
+typedef enum { LIQUIDITY_POOL_CONSTANT_PRODUCT = 0 } LiquidityPoolType;
+
+typedef struct {
+    Asset assetA;
+    Asset assetB;
+    int32_t fee;  // Fee is in basis points, so the actual rate is (fee/100)%
+} LiquidityPoolConstantProductParameters;
+
+typedef struct {
+    LiquidityPoolType type;
+    union {
+        LiquidityPoolConstantProductParameters
+            constantProduct;  // type == LIQUIDITY_POOL_CONSTANT_PRODUCT
+    };
+} LiquidityPoolParameters;
+
+typedef struct {
+    AssetType type;
+    union {
+        AlphaNum4 alphaNum4;
+        AlphaNum12 alphaNum12;
+        LiquidityPoolParameters liquidityPool;
+    };
+} ChangeTrustAsset;
+
+typedef struct {
+    AssetType type;
+    union {
+        AlphaNum4 alphaNum4;
+        AlphaNum12 alphaNum12;
+        uint8_t liquidityPoolID[32];
```

Comment:
Maybe you could defined `LIQUIDITY_POOL_ID_SIZE` to avoid using `32` in various places.